### DISCUSSION TITLE
[SSG] How to generate a json file as part of the Next build?

### DISCUSSION THREAD

**Discussion by marceloverdijk** [vercel/next.js#19587] [Help]
For a static generated site I would like to generate a Fuse.js `fuse-index.json` Next.js during build time.
The Fuse search index will contain documents I can retrieve similar as I would retrieve data for normal pages using `getStaticProps`.

Is it possible to generate generate a SSG `fuse.index.json` 'page?' so it is part of the generated site as a static asset?
Or is there another api/hook when using Next.js to generate such a file?

<hr>

PS: this is how the Fuse.js index is generated:

```
// In the build step.
// Get the data.
const books = // Get the books. This would contain same code as I would do in getStaticProps for other pages. 

// Create the Fuse index.
const myIndex = Fuse.createIndex(['title', 'author.firstName'], books)

// Serialize and save it
fs.writeFile('fuse-index.json', JSON.stringify(myIndex.toJSON()))
```

**Comment by Tomburgs** [comment]
I'm not sure if I understand correctly, but it seems like you are just trying to generate a json file which you just want to be able to access as a static asset?

All files in the `.next/static` directory will be automatically accessible under the `/_next/static` route, so you could just launch a new process during build time and output your files there, I've done a similar thing, you can have a look [here for an example](https://github.com/Tomburgs/pwa-boilerplate/blob/master/next.config.js#L35).

**Reply by marceloverdijk** [reply]
Thx @Tomburgs for your feedback, yes it is indeed similar as generating a sitemap.
For the data I process, and create the JSON for, I have nice TypeScript types for using Prisma. Could I use that using the `next.config.js` approach in your example?

**Reply by nickluger** [reply]
You cannot use TS in `next.config.js`, as stated [here](https://github.com/vercel/next.js/issues/5318#issuecomment-425398180). You could do it in a prebuild-step like `"build": "ts-node generateFuseIndex.ts && next build"` in your `package.json`. And in case you're using Vercel, you can customize the build command you use for deployment.

**Reply by marceloverdijk** [reply]
Oke then it's best to this before the `next build` indeed. I thought there was maybe a Next.js hook for this.
